tcms/rpc/api/bug.py

```diff
@@ -34,14 +34,16 @@ def details(url, **kwargs):
         :rtype: dict
     """
     result = cache.get(url)
-    if not result:
-        request = kwargs.get(REQUEST_KEY)
-        tracker = tracker_from_url(url, request)
-        if not tracker:
-            return None
-        result = tracker.details(url)
-        cache.set(url, result)
+    if result:
+        return result
 
+    request = kwargs.get(REQUEST_KEY)
+    tracker = tracker_from_url(url, request)
+    if not tracker:
+        return None
+
+    result = tracker.details(url)
+    cache.set(url, result)
     return result
```

Comment:
This is probably  a very minor improvement on the code climate score and can go together with the other 2 commits in master and their respective tests.